test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs

```diff
@@ -212,10 +347,46 @@ private static List<MetricPoint> GetMetricPoints(Metric metric)
         return metricPoints;
     }
 
+    private static void AssertMetricPoints(
+        List<MetricPoint> metricPoints,
+        List<string> expectedRoutes,
+        int expectedTagsCount,
+        bool validateNewSemConv = false,
+        bool validateOldSemConv = false)
+    {
+        // Assert that one MetricPoint exists for each ExpectedRoute
+        foreach (var expectedRoute in expectedRoutes)
+        {
+            MetricPoint? metricPoint = null;
+
+            foreach (var mp in metricPoints)
+            {
+                foreach (var tag in mp.Tags)
+                {
+                    if (tag.Key == SemanticConventions.AttributeHttpRoute && tag.Value.ToString() == expectedRoute)
+                    {
+                        metricPoint = mp;
+                    }
+                }
+            }
+
+            if (metricPoint.HasValue)
+            {
+                AssertMetricPoint(metricPoint.Value, expectedRoute, expectedTagsCount, validateNewSemConv, validateOldSemConv);
+            }
+            else
+            {
+                Assert.Fail($"A metric for route '{expectedRoute}' was not found");
+            }
+        }
+    }
+
     private static KeyValuePair<string, object>[] AssertMetricPoint(
         MetricPoint metricPoint,
         string expectedRoute = "api/Values",
-        int expectedTagsCount = StandardTagsCount)
+        int expectedTagsCount = StandardTagsCount,
+        bool validateNewSemConv = false,
```

Comment:
This still won't allow us to just delete code without refactoring :)

Could we split this in to two? I do not see code reuse anyways in this method.

Reply:
done